### PR TITLE
Update Dependencies for Security Advisories

### PR DESCRIPTION
**Pull request by zoracon**
- https://github.com/certbot/certbot/security/dependabot/71
- https://github.com/certbot/certbot/security/dependabot/72

### CODE REVIEW

File: tools/pinning/current/pyproject.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ license = "Apache License 2.0"
 python = "^3.8"
 
 # workaround for: https://github.com/python-poetry/poetry-plugin-export/issues/183
-urllib3 = ">=1.25.4,<1.27"
```
Comment (zoracon) [inline tools/pinning/current/pyproject.toml:12]: Not sure if the work around still applies for this update. Let me know if I should amend this version set up.

Comment (bmw) [reply tools/pinning/current/pyproject.toml:12]: yeah i think we should still set a maximum `urllib3` version as that poetry-plugin-export issue is unresolved

there's more context at https://github.com/certbot/certbot/issues/9862#issuecomment-2163743729

**Comment by bmw** [comment]
the usual way we update dependencies is leaving this file unmodified and running `tools/pinning/current/repin.sh`. i think we should probably do that here

**Comment by bmw** [comment]
to add a bit more context, our dependency setup in this repo is anything but standard (read https://eff-certbot.readthedocs.io/en/latest/contributing.html#certbot-s-dependencies and the referenced files like `tools/pinning/DESIGN.md` if you decide you REALLY care), but tooling for supporting client side python applications in a wide variety of environments isn't/wasn't great

undoing the changes here, activating your developer virtual environment, and running `tools/pinning/current/repin.sh` should create the changes we want here tho

**Comment by zoracon** [comment]
> to add a bit more context, our dependency setup in this repo is anything but standard (read https://eff-certbot.readthedocs.io/en/latest/contributing.html#certbot-s-dependencies and the referenced files like `tools/pinning/DESIGN.md` if you decide you REALLY care), but tooling for supporting client side python applications in a wide variety of environments isn't/wasn't great
> 
> undoing the changes here, activating your developer virtual environment, and running `tools/pinning/current/repin.sh` should create the changes we want here tho

I will do that this week.  I had this setup on my old machine and forgot that this was the way since it's been a while. I will reset this on my new work laptop.

**Comment by bmw** [comment]
so now I think the problem is pyopenssl, the dependency that Will was working on ripping out, deprecated functionality that we're using in josepy. those APIs are raising warnings causing our tests to fail

i think the solution for now is to pin back pyopenssl in `tools/pinning/current/pyproject.toml` and run the repinning script again which will generate pinnings with an older version of pyopenssl. we'd then also need to make sure we stay on top of actually fixing this problem

i can create PRs and/or issues for this if you want. just let me know!

**Comment by bmw** [comment]
actually, it looks like adrien already did this in https://github.com/certbot/certbot/pull/9993. i can work with him to get that merged if ya want and then i don't think we'd need this PR

**Comment by zoracon** [comment]
> actually, it looks like adrien already did this in #9993. i can work with him to get that merged if ya want and then i don't think we'd need this PR

That works for me. I will close this out and bookmark adrein's PR

**Pull request closed without merging**